test/CMakeLists.txt

```diff
@@ -33,9 +37,11 @@ endif()
 if(CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Intel)
     add_compile_options(-Wunused-parameter -Wextra -Wreorder -Wconversion -Wsign-conversion)
 
-    CHECK_CXX_COMPILER_FLAG(-march=native HAS_MARCH_NATIVE)
-    if (HAS_MARCH_NATIVE)
-        add_compile_options(-march=native)
+    if(NOT CMAKE_CXX_FLAGS MATCHES "-march")
+        CHECK_CXX_COMPILER_FLAG(-march=native HAS_MARCH_NATIVE)
+        if (HAS_MARCH_NATIVE)
+            add_compile_options(-march=native)
+        endif()
```

Comment:
Same question here, is there a need for building the tests of xtl with another arch than the host one?

Reply:
Yes, we build our software for distribution to customers computers. Everything should run on a certain minimum hardware spec (that changes every once in a while, currently its Haswell).

Reply:
BTW, I think we did the same change some year ago in xtensor and xtensor-blas.